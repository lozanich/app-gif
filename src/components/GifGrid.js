import React, { useState, useEffect } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "../components/GifGridItem";
import PropTypes from "prop-types";

function GifGrid({ category }) {
  const { data, loading } = useFetchGifs(category);
  const [images, setImages] = useState(data);

  useEffect(() => {
    setImages(data);
  }, [data]);

  const deleteImage = (id) => {
    setImages(images.filter((i) => i.id !== id));
  };

  return (
    <>
      <h2 className="animate__fadeInDown">{category}</h2>
      {loading && (
        <div className="animate__animated animate__flash">Cargando</div>
      )}

      <div>
        <ol className="card-grid">
          {images.map((item) => {
            return (
              <GifGridItem key={item.id} {...item} deleteImageG={deleteImage} />
            );
          })}
        </ol>
      </div>
    </>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
