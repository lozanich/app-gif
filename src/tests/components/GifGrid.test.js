import "@testing-library/jest-dom";
import React from "react";
import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente GifGrid", () => {
  const category = "Goku";

  test("Deberia de ser el snapsshot de GifGrid", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Deberia de mostrar items cuando cargan imagenes use Fetch gifs", () => {
    const imgs = [
      {
        id: "278anas",
        title: "Una imagen",
        url: "https://",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(imgs.length);
  });
});
