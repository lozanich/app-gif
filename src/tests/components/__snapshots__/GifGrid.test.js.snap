// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Deberia de mostrar items cuando cargan imagenes use Fetch gifs 1`] = `
<Fragment>
  <h2
    className="animate__fadeInDown"
  >
    Goku
  </h2>
  <div>
    <ol
      className="card-grid"
    >
      <GifGridItem
        deleteImageG={[Function]}
        id="278anas"
        key="278anas"
        title="Una imagen"
        url="https://"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Deberia de ser el snapsshot de GifGrid 1`] = `
<Fragment>
  <h2
    className="animate__fadeInDown"
  >
    Goku
  </h2>
  <div
    className="animate__animated animate__flash"
  >
    Cargando
  </div>
  <div>
    <ol
      className="card-grid"
    />
  </div>
</Fragment>
`;
